### PR TITLE
PTX-23350 validatePxPodsAnnotations for StorageNode should be called when op >= 23.10.0

### PR DESCRIPTION
**Pull request by nilanto07-px**
Some `storageNode.Status.NodeAttributes` like `nodeAttributes` were introduced in op 23.10.0
This PR is to check for these attributes only when opVersion is higher than expected. 

**Which issue(s) this PR fixes** (optional)
Closes # PTX-23350

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -894,7 +899,7 @@ func validatePxPodsAnnotations(cluster *corev1.StorageCluster, timeout, interval
 			// This should validate the annotation cluster-autoscaler.kubernetes.io/safe-to-evict value,
 			// when node is Storage, annotation value should be set to false,
 			// when node is Storageless, annotation value should be set to true
-			if storageNode.Status.NodeAttributes.Storage != nil && *storageNode.Status.NodeAttributes.Storage != !safeToEvict {
+			if opVersion.GreaterThanOrEqual(OpVer23_10_3) && storageNode.Status.NodeAttributes.Storage != nil && *storageNode.Status.NodeAttributes.Storage != !safeToEvict {
```
Comment (nikolaypopov) [inline pkg/util/test/util.go:902]: This is not gonna work, we should either find a different way to validate this or skip the whole thing all together for old operator and just print something like PX Operator version is too old and we do not support validating this

Comment (nikolaypopov) [reply pkg/util/test/util.go:902]: Also next debug print after this statement will not be useful as the older PX Operators do not have `*storageNode.Status.NodeAttributes.Storage`

Comment (nilanto07-px) [reply pkg/util/test/util.go]: Addressed the comments.

**Review by nikolaypopov — APPROVED** [review]

File: pkg/util/test/util.go [hunk 2]
```diff
@@ -867,6 +868,15 @@ func ValidateStorageCluster(
 
 func validatePxPodsAnnotations(cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
 	logrus.Debug("Validate PX pods..")
+	opVersion, err := GetPxOperatorVersion()
+	if err != nil {
+		logrus.Errorf("Operator version not found!")
+		return err
+	}
+	if opVersion.GreaterThanOrEqual(opVer23_10) {
```
Comment (nikolaypopov) [inline pkg/util/test/util.go:876]: This is wrong statement, it needs to be lower then 23.10 and only then we skip it, 23.10.0 or higher is good and we should not skip it

**Review by nikolaypopov — CHANGES_REQUESTED** [review]

File: pkg/util/test/util.go (hunk 2 above)
Comment (nilanto07-px) [reply pkg/util/test/util.go]: Ah. Updated.

**Comment by nilanto07-px** [comment]
https://jenkins.pwx.dev.purestorage.com/job/Torpedo/view/Torpedo%20NextPX/job/tp-ocp-op-px-upgrade/83/console 
Fixed worked. 
check around: 
`15:32:10 time="2024-04-16T10:03:16Z"
15:32:32 time="2024-04-16T10:03:39Z"`

**Review by nikolaypopov — APPROVED** [review]
LGTM

**Pull request merged**